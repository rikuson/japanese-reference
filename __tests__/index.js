// @flow

/* eslint-disable import/no-extraneous-dependencies */
import fs from "fs";
import path from "path";

import globby from "globby";

it("should build example correctly", () => {
  const testFolder = __dirname + "/../dist";
  const files = globby.sync("**/*", {
    cwd: testFolder,
    nodir: true,
    dot: true,
  });

  // should have html files
  const htmlFiles = files.filter(file => file.endsWith(".html"));
  expect(htmlFiles).toMatchSnapshot();

  const jsonApiFiles = files.filter(
    file => file.startsWith("phenomic") && file.endsWith(".json"),
  );
  // should have matching json files
  expect(jsonApiFiles).toMatchSnapshot();

  // should have assets
  const assetsBundlerFiles = files.filter(
    file =>
      !htmlFiles.includes(file) &&
      !jsonApiFiles.includes(file) &&
      file.startsWith("phenomic"),
  );
  expect(assetsBundlerFiles.length).toBe(2);
  expect(
    files.filter(
      file =>
        !htmlFiles.includes(file) &&
        !jsonApiFiles.includes(file) &&
        !assetsBundlerFiles.includes(file),
    ),
  ).toMatchSnapshot();
});

it("should make dynamic pages with pagination", () => {
  const file1 = fs.readFileSync(
    path.join(__dirname, "..", "dist", "repositories", "index.html"),
    { encoding: "utf8" },
  );
  expect(file1).toContain('<div class="PageRepositories-repo">');
  expect(file1).not.toContain("ActivityIndicator");

  const file2 = fs.readFileSync(
    path.join(
      __dirname,
      "..",
      "dist",
      "repositories",
      "page",
      "2",
      "index.html",
    ),
    { encoding: "utf8" },
  );
  expect(file2).toContain('<div class="PageRepositories-repo">');
  expect(file2).not.toContain("ActivityIndicator");
});
