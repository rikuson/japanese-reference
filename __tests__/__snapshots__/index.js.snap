// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should build example correctly 1`] = `
Array [
  "404.html",
  "about/index.html",
  "blog/after/c2Vjb25kLXBvc3Q=/index.html",
  "blog/after/dGhpcmQtcG9zdA==/index.html",
  "blog/after/Zm91cnRoLXBvc3Q=/index.html",
  "blog/after/ZmlmdGgtcG9zdA==/index.html",
  "blog/after/Zmlyc3QtcG9zdA==/index.html",
  "blog/fifth-post/index.html",
  "blog/first-post/index.html",
  "blog/fourth-post/index.html",
  "blog/index.html",
  "blog/second-post/index.html",
  "blog/third-post/index.html",
  "index.html",
  "repositories/index.html",
  "repositories/page/2/index.html",
  "repositories/page/3/index.html",
  "repositories/page/4/index.html",
]
`;

exports[`should build example correctly 2`] = `
Array [
  "phenomic/content/pages/item/.json",
  "phenomic/content/pages/item/about.json",
  "phenomic/content/posts/by-default/1/desc/date/limit-4.json",
  "phenomic/content/posts/by-default/1/desc/date/limit-6.json",
  "phenomic/content/posts/by-default/1/desc/date/limit-6/after-c2Vjb25kLXBvc3Q=.json",
  "phenomic/content/posts/by-default/1/desc/date/limit-6/after-dGhpcmQtcG9zdA==.json",
  "phenomic/content/posts/by-default/1/desc/date/limit-6/after-Zm91cnRoLXBvc3Q=.json",
  "phenomic/content/posts/by-default/1/desc/date/limit-6/after-ZmlmdGgtcG9zdA==.json",
  "phenomic/content/posts/by-default/1/desc/date/limit-6/after-Zmlyc3QtcG9zdA==.json",
  "phenomic/content/posts/item/fifth-post.json",
  "phenomic/content/posts/item/first-post.json",
  "phenomic/content/posts/item/fourth-post.json",
  "phenomic/content/posts/item/second-post.json",
  "phenomic/content/posts/item/third-post.json",
]
`;

exports[`should build example correctly 3`] = `
Array [
  "favicon.ico",
  "robots.txt",
]
`;
